lib/cn/route.c

```diff
@@ -319,7 +334,7 @@ route_node_next(struct route_node *node)
         return NULL;
 
     next = rb_next(&node->rtn_node);
-    return rb_entry(next, struct route_node, rtn_node);
+    return next ? rb_entry(next, struct route_node, rtn_node) : NULL;
```

Comment:
That must have been fun to chase down!

Reply:
Fortunately I just noticed it. Didn't have to debug my way here :)